frame/democracy/src/lib.rs

```diff
@@ -1560,6 +1703,36 @@ impl<T: Config> Pallet<T> {
 		// `Compact<u32>`.
 		decode_compact_u32_at(&<DepositOf<T>>::hashed_key_for(proposal))
 	}
+
+	// Return a proposal of an index.
+	fn proposal(index: PropIndex) -> Result<(PropIndex, BoundedCallOf<T>, T::AccountId), Error<T>> {
+		PublicProps::<T>::get()
+			.into_iter()
+			.find(|(prop_index, _, _)| prop_index == &index)
+			.ok_or(Error::<T>::ProposalMissing)
+	}
+
+	/// Clear metadata, if `Some` and unrequest associated preimage.
+	fn clear_metadata(owner: MetadataOwner) {
+		if let Some(hash) = MetadataOf::<T>::take(owner.clone()) {
```

Comment:
Why clone as opposed to a reference?
```suggestion
		if let Some(hash) = MetadataOf::<T>::take(&owner) {
```